src/Simplify.cpp

```diff
@@ -1449,7 +1449,9 @@ class Simplify : public IRMutator {
         } else if (is_one(b)) {
             expr = a;
         } else if (equal(a, b) &&
-                   !is_zero(b)) {
+                   !is_zero(b) &&
+                   // (n/0)/(n/0) should not be simplified to 1
+                   !(div_a && is_zero(div_a->b))) {
```

Comment:
This seems like we are handling only one of many possible cases here. Surely, there are plenty of indeterminate expressions e such that e/e should not be simplified to 1. It's not just division either: n/0 - n/0 should not simplify to 0.

I think maybe we should have a new intrinsic (similar to `signed_integer_overflow`), i.e. `indeterminate_expression`. Division by zero would simplify to this indeterminate expression, and then indeterminate expressions should not simplify away. If division by zero is the primary use case, then maybe it should just be `division_by_zero`.